### PR TITLE
Add JNI Support for Multi-line Delimiters and Include Test

### PR DESCRIPTION
**Pull request by SurajAralihalli**
## Description
This PR introduces the necessary changes to the cuDF jni to support the issue described in [NVIDIA/spark-rapids#11554](https://github.com/NVIDIA/spark-rapids/issues/11554). For further information, refer to the details in the [comment](https://github.com/NVIDIA/spark-rapids/issues/11554#issuecomment-2425327183).

Issue #15961 adds support for handling multiple line delimiters. This PR extends that functionality to JNI, which was previously missing, and also includes a test to validate the changes.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/RegexFlag.java [hunk 1]
```diff
@@ -28,7 +28,8 @@ public enum RegexFlag {
   DEFAULT(0),   // default
   MULTILINE(8), // the '^' and '$' honor new-line characters
   DOTALL(16),   // the '.' matching includes new-line characters
-  ASCII(256);   // use only ASCII when matching built-in character classes
+  ASCII(256),   // use only ASCII when matching built-in character classes
+  EXT_NEWLINE(512); // new-line matches extended characters
```
Comment (mythrocks) [inline java/src/main/java/ai/rapids/cudf/RegexFlag.java:32]: A short blurb here, indicating the definition of the behaviour with `extended characters` would be useful here, for the uninitiated.

The blurb should probably include the fact that when `EXT_NEWLINE` is used, the following characters would be treated as equivalent to a newline:
1. `NEW_LINE('\n')`
2. `CARRIAGE_RETURN('\r)`
3. ...

**Review by mythrocks — COMMENTED** [review]
LGTM, except for documentation of the new behaviour.

File: java/src/main/java/ai/rapids/cudf/RegexFlag.java (hunk 1 above)
Comment (SurajAralihalli) [reply java/src/main/java/ai/rapids/cudf/RegexFlag.java]: Done!

**Review by mythrocks — APPROVED** [review]
LGTM!

**Review by revans2 — APPROVED** [review]

**Pull request merged**